climpred/classes.py

```diff
@@ -376,22 +405,26 @@ def _apply_func(self, func, *args, **kwargs):
         datasets = self._datasets.copy()
 
         # More explicit than nested dictionary comprehension.
-        for outer_k, outer_v in datasets.items():
-            # If initialized, control, uninitialized and just a singular
-            # dataset, apply the function directly to it.
-            if isinstance(outer_v, xr.Dataset):
-                datasets.update({outer_k: func(outer_v, *args, **kwargs)})
-            else:
-                # If a nested dictionary is encountered (i.e., a set of
-                # observations) apply to each individually.
-                #
-                # Similar to the ``add_observations`` method, this only seems to
-                # avoid inplace operations by copying the nested dictionary
-                # separately and then updating the main dictionary.
-                temporary_dataset = self._datasets[outer_k].copy()
-                for inner_k, inner_v in temporary_dataset.items():
```

Comment:
the douple loop was not needed anymore. needed before when observations was dict